api/src/main/java/net/kyori/adventure/text/event/HoverEvent.java

```diff
@@ -244,6 +247,9 @@ public ShowItem(final @NonNull Key item, final @NonNegative int count, final @Nu
      * @return a {@code ShowItem}
      */
     public @NonNull ShowItem count(final @NonNegative int count) {
+      if(count == this.count) {
+        return this;
+      }
```

Comment:
```suggestion
      if(count == this.count) return this;
```
applies to all cases like this.